flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamContextEnvironment.java

```diff
@@ -57,9 +58,12 @@ public JobExecutionResult execute(StreamGraph streamGraph) throws Exception {
 		// running from the CLI will override the savepoint restore settings
 		jobGraph.setSavepointRestoreSettings(ctx.getSavepointRestoreSettings());
 
-		JobExecutionResult jobExecutionResult =  ctx.getClient()
-			.submitJob(jobGraph, ctx.getUserCodeClassLoader())
-			.getJobExecutionResult();
+		JobExecutionResult jobExecutionResult;
+		if (ctx.isDetached()) {
+			jobExecutionResult = new DetachedJobExecutionResult(ClientUtils.submitJob(ctx.getClient(), jobGraph).getJobID());
```

Comment:
You do not need to re-wrap it in a `DetachedJobExecutionResult`.

Reply:
The same as above.